src/thor/map_matcher.cc

```diff
@@ -155,14 +155,19 @@ MapMatcher::FormPath(meili::MapMatcher* matcher,
       // get the timezone
       nodeinfo = tile->node(directededge->endnode());
       const auto* tz = DateTime::get_tz_db().from_index(nodeinfo->timezone());
+      if (!tz)
+        continue;
       // if its timestamp based need to signal that out to trip leg builder
-      if (options.shape(0).time() != -1.0) {
+      if (!options.shape(0).has_date_time() && options.shape(0).time() != -1.0) {
```

Comment:
we only use the timestamp if we dont have a date string. this is important as on the outside of this, when we are doing legs, we set the date_time on the legs origins. (this is still a TODO but this code will depend on it)